dragonfly/actions/sendinput.py

```diff
@@ -68,10 +85,16 @@ class KeyboardInput(Structure):
                      win32con.VK_RWIN,
                     )
 
-    def __init__(self, virtual_keycode, down):
-        scancode = windll.user32.MapVirtualKeyA(virtual_keycode, 0)
+    def __init__(self, virtual_keycode, down, scancode=-1):
+        """Initialize structure based on key type."""
+        if scancode == -1:
+            scancode = windll.user32.MapVirtualKeyW(virtual_keycode, 0)
```

Comment:
Can you just check if this returns 0 instead of keeping a list of SOFT_KEYS? That's what the documentation implies -- but maybe it doesn't work.

Reply:
Honestly, this is exactly what I should've done, but I never tested it. I'll add it to the list of things to check up on. There might've been a good reason I did it this way, but I can't remember.

Reply:
On further consideration, I think we are actually best off only setting the scancode when in Unicode mode, to forward Unicode characters. Otherwise, we can just use the virtual key like we used to. I think that's a bit cleaner because the scan code is actually supposed to be a hardware-dependent code whereas the virtual key is higher-level and closer to what we want anyways (based on https://docs.microsoft.com/en-us/windows/desktop/inputdev/about-keyboard-input). It also just means less code and conversion here to deal with (no need to ever call MapVirtualKeyW here).